controllers/machinedeploymentupgrade_controller.go

```diff
@@ -57,3 +121,73 @@ func (r *MachineDeploymentUpgradeReconciler) SetupWithManager(mgr ctrl.Manager)
 		For(&anywherev1.MachineDeploymentUpgrade{}).
 		Complete(r)
 }
+
+func (r *MachineDeploymentUpgradeReconciler) reconcile(ctx context.Context, log logr.Logger, mdUpgrade *anywherev1.MachineDeploymentUpgrade) (ctrl.Result, error) {
+	log.Info("Upgrading all worker nodes")
+	for _, machineRef := range mdUpgrade.Spec.MachinesRequireUpgrade {
+		nodeUpgrade := nodeUpgrader(machineRef, "v1.28.3-eks-1-28-9", "v3.5.9-eks-1-28-9", false)
+		if err := r.client.Create(ctx, nodeUpgrade); client.IgnoreAlreadyExists(err) != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to create node upgrader for machine %s:  %v", machineRef.Name, err)
+		}
+	}
+
+	err := r.updateStatus(ctx, log, mdUpgrade)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *MachineDeploymentUpgradeReconciler) reconcileDelete(ctx context.Context, log logr.Logger, mdUpgrade *anywherev1.MachineDeploymentUpgrade) (ctrl.Result, error) {
+	log.Info("Reconcile MachineDeploymentUpgrade deletion")
+
+	log.Info("Deleting MachineDeploymentUpgrade", "Name", mdUpgrade.Name)
+	if err := r.client.Delete(ctx, mdUpgrade); err != nil {
+		return ctrl.Result{}, fmt.Errorf("deleting MachineDeploymentUpgrade: %v", err)
+	}
+
+	// Remove the finalizer on MachineDeploymentUpgrade object
+	controllerutil.RemoveFinalizer(mdUpgrade, mdUpgradeFinalizerName)
+	return ctrl.Result{}, nil
+}
+
+func patchMachineDeploymentUpgrade(ctx context.Context, patchHelper *patch.Helper, mdUpgrade *anywherev1.MachineDeploymentUpgrade, patchOpts ...patch.Option) error {
+	// Always attempt to patch the object and status after each reconciliation.
+	return patchHelper.Patch(ctx, mdUpgrade, patchOpts...)
+}
+
+func (r *MachineDeploymentUpgradeReconciler) updateStatus(ctx context.Context, log logr.Logger, mdUpgrade *anywherev1.MachineDeploymentUpgrade) error {
+	// When MachineDeploymentUpgrade is fully deleted, we do not need to update the status. Without this check
+	// the subsequent patch operations would fail if the status is updated after it is fully deleted.
+	if !mdUpgrade.DeletionTimestamp.IsZero() && len(mdUpgrade.GetFinalizers()) == 0 {
+		log.Info("MachineDeploymentUpgrade is deleted, skipping status update")
+		return nil
+	}
+
+	log.Info("Updating MachineDeploymentUpgrade status")
+
+	var ready int64
```

Comment:
Let's have all `var` declarations at the top of the function, otherwise do something like `ready := int64(0)`, tho I don't remember if that's how the syntax is.